packages/peregrine/docs/RestApi.md

```diff
@@ -0,0 +1,190 @@
+# REST API Clients
```

Comment:
I'm hesitant about introducing a large abstraction for REST API calls at this stage (our first integration to a non-GQL API). I'm sure people are getting tired of me beating this drum at this point, but I strongly believe that an abstraction should not be created over some other API until we've truly felt the pain of not having that abstraction. At that point, my preference is always going to be the minimal abstraction that covers the immediate use-cases.

For this particular abstraction, I have a few specific concerns.

1. This abstracts over the `fetch` API, but limits a consumer's ability to use other features exposed through `fetch`. Examples include having no control over the `credentials` option, the `headers` option, the `integrity` option, etc.

2. I don't see that we're using `addInterrupt` anywhere in this code. Once we've exposed an API we have to support it, and I'm not incredibly comfortable exposing an API we don't have a need for. There is a major hidden cost there.

I feel like I need help understanding what problem this `fetch` abstraction addresses in this specific PR. I've worked on several fairly large React/Redux CRUD applications, and I personally haven't seen the problems that this abstraction (I believe) is aiming to solve.

Reply:
_I'm not directly involved but I'm a strong believer in keeping things simple and I hate complex abstractions so_

From my previous experience at Gene: Magento already has _a lot_ of abstractions. They're taxing and confusing to learn and master, making the learning curve and time needed to work a new system much more confusion. On top of the complexity that PWA is already going to be, I'm strongly in favour of keeping things as simple as possible.

If we need this abstracted, for whatever reason, I feel like enough people use REST with Node that there is a community solution that is well understood and _may already be known_ by developers coming onto PWA (like maybe the one Microsoft wrote: https://github.com/Microsoft/typed-rest-client). Let's not continue the practice of Magento reinventing the wheel with everything we do.